packages/curve-ui-kit/src/shared/ui/Metric.tsx

```diff
@@ -0,0 +1,195 @@
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import { SizesAndSpaces } from 'themes/design/1_sizes_spaces'
+import { Fonts } from 'themes/typography/fonts'
+
+export const SIZES = ['small', 'medium', 'large', 'extraLarge'] as const
+type Size = (typeof SIZES)[number]
+
+// Correspond to flexbox align items values.
+export const ALIGNMENTS = ['start', 'center', 'end'] as const
+type Alignment = (typeof ALIGNMENTS)[number]
+
+type UnitOptions = {
+  symbol: string
+  position: 'prefix' | 'suffix'
+  abbreviate: boolean
+}
+
+const dollar: UnitOptions = {
+  symbol: '$',
+  position: 'prefix',
+  abbreviate: true,
+}
+
+const percentage: UnitOptions = {
+  symbol: '%',
+  position: 'suffix',
+  abbreviate: false,
+}
+
+export const UNITS = ['dollar', 'percentage'] as const
+type Unit = (typeof UNITS)[number] | UnitOptions
+
+const UNIT_MAP: Record<(typeof UNITS)[number], UnitOptions> = {
+  dollar,
+  percentage,
+} as const
+
+// I don't know if this is the right place or approach for size mapping.
+// Perhaps it's suited somewhere better in the design folder.
+const SizeMapping: Record<Size, string> = {
```

Comment:
we could just receive `variant: TypographyVariantKey` instead of a separate mapping